src/python_testing/TC_DeviceConformance.py

```diff
@@ -50,6 +50,22 @@ async def setup_class_helper(self):
         self.xml_device_types, problems = build_xml_device_types()
         self.problems.extend(problems)
 
+    def _get_device_type_id(self, device_type_name: str) -> int:
+        id = [id for id, dt in self.xml_device_types.items() if dt.name.lower() == device_type_name.lower()]
+        if len(id) != 1:
+            self.fail_current_test(f"Unable to find {device_type_name} device type")
+        return id[0]
+
+    def _has_nim(self):
+        nim_id = self._get_device_type_id('network infrastructure manager')
+        for endpoint in self.endpoints_tlv.values():
+            desc = Clusters.Descriptor
+            device_types = [dt.deviceType for dt in endpoint[desc.id][desc.Attributes.DeviceTypeList.attribute_id]]
+            if nim_id in device_types:
+                # TODO: it's unclear if this needs to be present on every endpoint. Right now, this assumes one is sufficient.
```

Comment:
Presence of only 1 is OK, since it's an application device type.

Reply:
The question was around multiple endpoints. So one endpoint with NIM one with ex. lightbulb. I'm not sure how the spec text is going to land around that condition.

Reply:
Given ACL is node scoped and the fact that we do not want to allow MACL for other device types than NIM, I would say that we have a requirement that NIM is not allowed to have other EP's with other device types 🤔

Reply:
Let's not throw the baby out with the bathwater. It makes no sense saying there can only be a single endpoint and it has to be NIM.

Maybe we need to update ACL TP instead and work an update on the feature conformance to add steps that it is only present if at least one endpoint is a device type that has it as O at least.

Reply:
This does this currently - allowed if at least one endpoint is marked as being a NIM. It does mean you can have a weird combo device, but you're probably right that it's unlikely to happen in practice.

Reply:
OK, I'm going to resolve this for now so I can get this code in. I've opened a spec issue here: https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/10294